homeassistant/components/frontier_silicon/media_player.py

```diff
@@ -186,6 +196,9 @@ def media_image_url(self):
         if not self._source_list:
             self._source_list = await fs_device.get_mode_list()
 
+        if not self._volume_steps:
+            self._volume_steps = await fs_device.get_volume_steps()
```

Comment:
Subtract one here so you do not have to do write it multiple times. Why is it subtracted btw?
Would it be useful to cast it to an int as well?
```suggestion
            self._volume_steps = await fs_device.get_volume_steps() - 1
```

Reply:
Thanks, I've made the suggested changes in the latest commit.

The -1 is because the API seems include 'zero' in the number of steps (e.g. if the range is 0-40 then get_volume_steps returns 41).   Therefore without it a max volume of 1.0 sends it out of range (which causes the API to set the volume back to zero).